src/test/java/org/opensearch/security/system_indices/DenyIndicesTests.java

```diff
@@ -0,0 +1,166 @@
+
+    private void setupDenyIndicesEnabledWithSsl(Boolean securedIndicesAdditionalControlenable) throws Exception {
+
+        Settings denyIndexSettings = Settings.builder()
+            .put(ConfigConstants.SECURITY_SYSTEM_INDICES_ADDITIONAL_CONTROL_ENABLED_KEY, securedIndicesAdditionalControlenable)
+            .putList(ConfigConstants.SECURITY_SYSTEM_INDICES_KEY, listOfIndexesToTest)
+            .put("plugins.security.ssl.http.enabled", true)
+            .put("plugins.security.ssl.http.keystore_filepath", FileHelper.getAbsoluteFilePathFromClassPath("node-0-keystore.jks"))
+            .put("plugins.security.ssl.http.truststore_filepath", FileHelper.getAbsoluteFilePathFromClassPath("truststore.jks"))
+            .put("path.repo", repositoryPath.getRoot().getAbsolutePath())
+            .build();
+
+        setup(
+            Settings.EMPTY,
+            new DynamicSecurityConfig().setConfig("config_system_indices.yml")
+                .setSecurityRoles("roles_system_indices.yml")
+                .setSecurityInternalUsers("internal_users_system_indices.yml")
+                .setSecurityRolesMapping("roles_mapping_system_indices.yml"),
+            denyIndexSettings,
+            true
+        );
+    }
+
+    /**
+     * Creates a set of test indices and indexes one document into each index.
+     *
+     * @throws Exception
+     */
+
+    private RestHelper keyStoreRestHelper() {
+        RestHelper restHelper = restHelper();
+        restHelper.keystore = "kirk-keystore.jks";
+        restHelper.enableHTTPClientSSL = true;
+        restHelper.trustHTTPServerCertificate = true;
+        restHelper.sendAdminCertificate = true;
+        return restHelper;
+    }
+
+    private RestHelper sslRestHelper() {
+        RestHelper restHelper = restHelper();
+        restHelper.enableHTTPClientSSL = true;
+        return restHelper;
+    }
+
+    /***************************************************************************************************************************
+     * Search api tests. Search is a special case.
+     ***************************************************************************************************************************/
+
+    private void validateSearchResponse(RestHelper.HttpResponse response, int expectecdHits) throws IOException {
+        assertEquals(RestStatus.OK.getStatus(), response.getStatusCode());
+
+        XContentParser xcp = XContentType.JSON.xContent()
+            .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getBody());
+        SearchResponse searchResponse = SearchResponse.fromXContent(xcp);
+        assertEquals(RestStatus.OK, searchResponse.status());
+        assertEquals(expectecdHits, searchResponse.getHits().getHits().length);
+        assertEquals(0, searchResponse.getFailedShards());
+        assertEquals(5, searchResponse.getSuccessfulShards());
+    }
+
+    @Test
+    public void testSearchWithDenyIndicesAsSuperAdmin() throws Exception {
```

Comment:
This name is confusing, seems like it is testSuperAdminAccessSystemIndices is what is being tested.  There should also be a second test that confirms the behavoir is the same with the feature on

Reply:
_"This name is confusing, seems like it is testSuperAdminAccessSystemIndices is what is being tested"_

It is testing DenyIndices. The comment  "        // search system indices" inside the method is a copy/past error.


_There should also be a second test that confirms the behavoir is the same with the feature on_

This test is with the feature on. 
I didn't create a separate test with the feature off because SuperAdmin should have access in both scenarios, so i just used the more stringent scenario.